infra/src/main/kotlin/fi/fta/geoviite/api/frameconverter/v1/FrameConverterRequestDataV1.kt

```diff
@@ -0,0 +1,121 @@
+package fi.fta.geoviite.api.frameconverter.v1
+
+import fi.fta.geoviite.infra.common.AlignmentName
+import fi.fta.geoviite.infra.common.Oid
+import fi.fta.geoviite.infra.common.TrackMeter
+import fi.fta.geoviite.infra.common.TrackNumber
+import fi.fta.geoviite.infra.error.InputValidationException
+import fi.fta.geoviite.infra.tracklayout.LayoutTrackNumber
+import fi.fta.geoviite.infra.tracklayout.LocationTrack
+
+fun createValidTrackMeterOrNull(
+    trackKilometer: Int?,
+    trackMeter: Int?,
+): Pair<TrackMeter?, List<FrameConverterErrorV1>> {
+    return if (trackKilometer == null || trackMeter == null) {
```

Comment:
Tällaiset null-käsittelyt funktioiden sisällä on minusta vähän anti-pattern. On parempi että funkkari ottaa non-null-tyypin ja sitä kutsutaan esim `param?.let(::funktioKutsu)`. Työntämällä null-checkit kutsuketjua ylöspäin saa yksinkertaisemmat funkkarit, poikkeustapaukset dokumentoituu koodiin sinne missä ne ovat oikeasti ymmärrettäviä ja usein moni check/null muuttuu turhaksikin. Toki tällaisessa missä on useampi nullable arg se tarkoitttaisi iffittelyä kutsupäässä jolloin voi tosiaan olla vaivan arvoista mutta noissa alemmissa 1-argumenttisissa erityisesti se näyttää hassulta.